src/utils/gridUtils.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { arrayIsntMxNErr, defaults, promotedHeadersCannotBeUsedWithoutAdjustingColumnNamesErr, unexpectedErr } from "../utils/constants";
+import { Grid, TableData } from "../types";
+
+interface MergedGridConfig {
+    promoteHeaders: boolean;
+    adjustColumnNames: boolean;
+}
+interface MergedGrid {
+    data: string[][];
+    config: MergedGridConfig;
+}
+
+const parseToTableData = (grid: Grid): TableData => {
+    const mergedGrid: MergedGrid = {
+        config: {
+            promoteHeaders: grid.config?.promoteHeaders ?? false,
+            adjustColumnNames: grid.config?.adjustColumnNames ?? true,
+        },
+        data: grid.data.map((row) => row.map((value) => value.toString())),
```

Comment:
What if the value is null, we had a fix that null become empty string

Reply:
Can you reproduce passing null with the type checks that exist?

Reply:
As far as I see parseToTableData run on a raw data, which can be null, it was easly reproduced using the sandbox by sending null value in the json.
Could you please double check \ add test

Reply:
See latest commit, fixed and changed tests environment to support this.